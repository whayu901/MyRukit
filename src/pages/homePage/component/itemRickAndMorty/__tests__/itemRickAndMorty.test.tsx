import { Alert } from "react-native";
import React from "react";
import renderer from "react-test-renderer";

import ItemRickAndMorty from "..";

jest.useFakeTimers();

// Note: test renderer must be required after react-native.

const mock = {
  id: 0,
  name: "wahyu",
  species: "human",
  status: "alive",
  type: "none",
  gender: "man",
  origin: {
    name: "none",
    url: "https",
  },
  location: {
    name: "none",
    url: "https",
  },
  image: "none",
  url: "non",
  created: "none",
  episode: ["hello", "world"],
};

describe("test ItemRickAndMorty", () => {
  describe("rendering", () => {
    it("should match to snapshot ItemRickAndMorty", () => {
      const temRickAndMorty = renderer.create(
        <>
          <ItemRickAndMorty
            indexItem={10}
            onPress={() => Alert.alert("hello")}
            item={mock}
          />
        </>,
      );
      expect({
        temRickAndMorty,
      }).toMatchSnapshot("ItemRickAndMorty snapshot true");
    });
  });
});
