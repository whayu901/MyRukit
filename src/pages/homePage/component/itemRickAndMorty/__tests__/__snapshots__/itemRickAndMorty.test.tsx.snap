// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemRickAndMorty rendering should match to snapshot ItemRickAndMorty: ItemRickAndMorty snapshot true 1`] = `
Object {
  "temRickAndMorty": <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "none",
          }
        }
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Lekton-Bold",
            "fontSize": 16,
          }
        }
      >
        wahyu
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Lekton-Regular",
            "fontSize": 12,
            "paddingTop": 8,
          }
        }
      >
        human
      </Text>
    </View>
  </View>,
}
`;
